test/e2e/must-gather/musth_gather_test.go

```diff
@@ -108,3 +142,45 @@ func checkfilesExist(listOfFiles []string, path string) error {
 	}
 	return nil
 }
+
+// Look for node yaml manifest in `folder` and return
+// all the names of all the nodes with
+// label "node-role.kubernetes.io/worker"
+func getWorkerNodesNames(folder string) ([]string, error) {
+	retval := []string{}
+	items, err := ioutil.ReadDir(folder)
+	if err != nil {
+		return retval, err
+	}
+
+	for _, item := range items {
+		if item.IsDir() {
+			continue
+		}
+
+		data, err := ioutil.ReadFile(filepath.Join(folder, item.Name()))
+		if err != nil {
+			return retval, err
+		}
+
+		node := &corev1.Node{}
+		err = yaml.Unmarshal(data, node)
+		if err != nil {
+			return retval, err
+		}
+
+		if _, ok := node.ObjectMeta.Labels["node-role.kubernetes.io/worker"]; ok {
+			retval = append(retval, node.Name)
+		}
+	}
+	return retval, nil
+}
+
+func map2String(orig []string, f func(string) string) []string {
```

Comment:
minor: guess this can be inlined (and simplified) in the call site?
Or, if the intent is to do something similar to https://pkg.go.dev/strings#Map , can be renamed to something like `MapStrings` ?

Reply:
My intention was to "hide" a detail in a function to make it easier to read.

Reply:
changed